### DISCUSSION TITLE
Using Typescript?

### DISCUSSION THREAD

**Discussion by cliffordfajardo** [c9fe/22120#105] [Ideas]
Hi @i5ik,
Would you be open to the idea of using Typescript?
- https://www.typescriptlang.org
I think it would help a lot with catching not only bugs before compile time, but also would make the javascript build system easier.
Currently you're using Webpack, which is fine, but if you use Typescript you would get optional types for free & web pack could be removed since the typescript compiler can transpire code as well

By the way, I was surprised to see how minimal the code is for the codebase, given the power of this tool 😄

**Comment by crisdosaygo** [comment]
Hey @cliffordfajardo 

> By the way, I was surprised to see how minimal the code is for the codebase, given the power of this tool 😄

Thank you for saying that! That's very nice.

I admire that too about this project. Lately I've considered it's even beefed up quite a lot. Before search was added, it was even more minimal!

Apart from my fantastic skill as a developer (😄) the main reason for this is because I leverage the power of [chrome remote debugging protocol](https://chromedevtools.github.io/devtools-protocol/) -- it's really very powerful what it can achieve. 

And basically I'm building on the back of the biggest browser, at a time when browsers are the most sophisticated and important software publishing platform.

So I think it's a really great idea, and as you have astutely pointed out: it's powerful and minimal.
 
I made it on a whim, I wanted to see if it was possible to save the intercepted requests to disk and replay them, and I felt quite inspired when I saw that this was possible. 

I encourage you to try to think of other inspiring things you could do with CRDP and try it out on your own. Because I think there's so much you could do--so many cool ideas waiting to be made! 💪🏻 ✊🏻 ⚡ :)
